cmd/bundle/summarise.go

```diff
@@ -0,0 +1,67 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/spf13/cobra"
+)
+
+func newSummariseCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "summarise",
+		Short: "Describe the bundle resources and their deployment states",
+
+		PreRunE: ConfigureBundleWithVariables,
+
+		// This command is currently intended for the Databricks VSCode extension only
+		Hidden: true,
+	}
+
+	var forcePull bool
+	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Skip local cache and load the state from the remote workspace")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		b := bundle.Get(cmd.Context())
+
+		err := bundle.Apply(cmd.Context(), b, phases.Initialize())
+		if err != nil {
+			return err
+		}
+
+		cacheDir, err := terraform.Dir(cmd.Context(), b)
+		if err != nil {
+			return err
+		}
+		_, err = os.Stat(filepath.Join(cacheDir, terraform.TerraformStateFileName))
+		noCache := errors.Is(err, os.ErrNotExist)
+
+		if forcePull || noCache {
+			err = bundle.Apply(cmd.Context(), b, terraform.StatePull())
+			if err != nil {
+				return err
+			}
+		}
+
+		err = bundle.Apply(cmd.Context(), b, terraform.Load())
```

Comment:
Instead of manually calculating the modified status during Load time, you can leverage terraform APIs to do so.

This would compute the terraform plan against the state file, without refreshing the state file (ie mostly locally).  This would also give you information whether the resource will be "updated" or "recreated" out of the box for free.
We would not have to manually calculate the diff.

We already have a terraform.Plan() mutator you can modify to add an option for this.
```
tf.Plan(ctx, tfexec.Refresh(false), tfexec.Out(outputPathForPlan))
plan, err := tf.ShowPlanFile(ctx, outputPathForPlan)
```

To see the modified status you can:
```
	for _, change := range plan.ResourceChanges {
		tfActions := change.Change.Actions
		if tfActions.Read() || tfActions.NoOp() {
			continue
		}

		var action string
		switch {
		case tfActions.Update():
			action = "Update"
		case tfActions.Create():
			action = "Create"
		case tfActions.Delete():
			action = "Delete"
		case tfActions.Replace():
			action = "Replace"
```

@pietern what do you think about leveraging the terraform API instead of relying on manually calculating the modified status?

Reply:
Checked with Pieter offline, we can go ahead with the current method and do the terraform APIs as a followup